sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/RequestHeadersSpyWireTest.java

```diff
@@ -216,6 +230,32 @@ public void readItemWithMaxIntegratedCacheStalenessInNegative() {
             .hasMessage("MaxIntegratedCacheStaleness duration cannot be negative");
     }
 
+    @Ignore // This test has to be run against sqlx endpoint
+    @Test(groups = { "simple" }, timeOut = TIMEOUT, dataProvider = "cacheBypassValues")
+    public void readItemWithCacheBypass(boolean cacheBypass) {
+        DedicatedGatewayRequestOptions dedicatedGatewayRequestOptions = new DedicatedGatewayRequestOptions();
+        dedicatedGatewayRequestOptions.setMaxIntegratedCacheStaleness((Duration.ofMillis(500)));
+        dedicatedGatewayRequestOptions.setBypassIntegratedCache(cacheBypass);
+        CosmosItemRequestOptions cosmosItemRequestOptions = new CosmosItemRequestOptions();
+        cosmosItemRequestOptions.setDedicatedGatewayRequestOptions(dedicatedGatewayRequestOptions);
+
+        String documentLink = getDocumentLink();
+
+        client.clearCapturedRequests();
+
+        RequestOptions requestOptions = ModelBridgeInternal.toRequestOptions(cosmosItemRequestOptions);
+        requestOptions.setPartitionKey(new PartitionKey(DOCUMENT_ID));
+        ResourceResponse<Document> response = client.readDocument(documentLink, requestOptions).block();
+        if (cacheBypass) {
+            String responseHeader = response.getResponseHeaders().get("x-ms-cosmos-cache-bypass");
```

Comment:
Synced up offline with @balalakshmin. `x-ms-cosmos-cache-bypass` response from the service / gateway denotes whether the request hit the cache or not.